### ISSUE TITLE
Lutron Occupancy Sensor Not Detected

### ISSUE THREAD

**Issue by czoog** [upsert/lutron-caseta-pro#6]
Hi Upsert,

I am not sure if this is a true issue yet, but I thought I would log it anyway.  I just added a  a ceiling mount occupancy sensor (Lutron LRF2-OCR2B-P-WH) to my radioRA 2 Select main repeater.  Currently that is ONLY thing connected to the system.  I have nothing else , no switches, nothing but the sensor, and perhaps that is the issue (i am still waiting for my switches to arrive).  The sensor shows up in the app and is assign to a room, but it is NOT set to control anything yet, obviously.

My debug logs show that the component sees the bridge and attempts to parse the devices on that bridge, but comes up with none (see log snippet).  The integration report shows the sensor as ID 2 and has a name and an area, but it does not have any "buttons" which i think is what is causing the issue. casetify.py seems to need to have the words "buttons" in the if else statement that identify the bridge, light/switchs, and remote.  Might simply need a catch for the occupancy sensors, which don't see to have buttons.  I've included my integration report as well, so you can see how the sensors are formatted.

I'll report back once I have a switch or two connected.  Maybe buttons will show up!

Thanks again for this awesome component!

[report.txt](https://github.com/upsert/lutron-caseta-pro/files/1917651/report.txt)

[ha_log_snippet.txt](https://github.com/upsert/lutron-caseta-pro/files/1917649/ha_log_snippet.txt)

**Comment by upsert** [comment]
Interesting. Maybe if it is not under Zones and has no Buttons, assume an occupancy sensor. Worse case I will have to do a `binary_sensor: [ 1, 2, 3]` type of config override.

I am still waiting on mine to ship, but once I get one I plan to integrate it as a new platform for the component. Should be a [Binary Sensor](https://www.home-assistant.io/components/binary_sensor/) of type 'occupancy'.

**Comment by upsert** [comment]
Sadly, I don't think I will be able to support the occupancy sensors as they do not output anything to telnet interface. I initially thought it was working, but it turns out I was just seeing the light level of the fixture associated with the occupancy sensor, i.e. the light in the assigned area.

Unless they update the telnet interface to support DEVICE output for Occupied/Unoccupied (page 140 of the [Integration Protocol](https://www.lutron.com/technicaldocumentlibrary/040249.pdf)), I'm not sure what else can be done. Maybe with the increasing use of Ra2 Select, the official integration partners will press for this to be added. The addition of occupancy sensors to Ra2 Select is a relatively new feature (March 2018).

**Comment by czoog** [comment]
Bummer. :(  Looks like these guys are trying to figure it out too.  [link](https://forums.lutron.com/showthread.php/4302-Occupancy-sensor-LRF2-OCR2B-needs-to-be-paired-with-a-light-for-integration-events)

One guy said he could see the device output using a serial interface

**Comment by upsert** [comment]
Yeah that makes sense. The last two posts are from people with the full RadioRA 2 Main Repeater and not RA2 Select. The RadioRA 2 Main Repeater has both a telnet and serial line interface and no doubt a full implementation of the Lutron Integration Protocol.

RA2 Select is based on their newer platform so I am still hoping one of their official integration partners like Control4 will want occupancy sensor triggers to work in their driver and Lutron will add it. So far they have announced Control4 support [here](https://forums.lutron.com/showthread.php/3739-Ra2-Select-Control4-drivers?p=19926&viewfull=1#post19926) and I found a driver [here](https://drivers.control4.com/solr/drivers/browse?&fq=manufacturer%3A%22Lutron%22&q=&start=38) which definitely uses Telnet for integration. The driver documentation is dated March 30, 2018 so maybe once more integrators start using it, they will feedback to Lutron about the missing output for occupancy sensor events.

I will keep an eye out for firmware updates to the bridge. Their last update was April 16, 2018.

**Comment by czoog** [comment]
Yup, let's hope they push lutron to add it.  The occupancy sensors are really nice and super responsive.  I would love to use more of them.

**Comment by upsert** [comment]
Firmware 06.05.14f000 is out and sadly it does not appear to change the telnet interface to have output events for occupancy sensors on RA2 Select.

Maybe we can appeal to Lutron engineers like @Aditya90 on Github to sneak the feature request into their backlog? 😸

**Comment by upsert** [comment]
Firmware 06.07.07f000 is out. No changes found with respect to the telnet interface.

**Comment by upsert** [comment]
Firmware 06.08.04f000 is out. No changes for the telnet interface. Probably preparing for [Caseta fan controller](https://fcc.report/FCC-ID/JPZ0119) I spotted a couple of weeks ago.

**Comment by upsert** [comment]
Firmware 06.09.01f000 is out. No changes found with respect to the telnet interface.

**Comment by upsert** [comment]
Firmware 06.09.05f000 is out. No changes to occupancy sensors in RA2 Select. Likely the firmware required to bump Caseta up to 75 devices.

**Comment by upsert** [comment]
Probably adding this too: https://fccid.io/JPZ0122. A new Radio RA2 / RA2 Select dimmer. Model RRD-PRO, 120 V, 250 W LED/CFL load. Looks like the two-wire RRD-6CL but with 250 W LED and phase adaptive like the RRD-6NA.

Apparently they have an [Advanced Programming Mode](http://www.lutron.com/TechnicalDocumentLibrary/048703.pdf) too where you can adjust the fade times, delayed fade to off, disable the LEDs and adjust trim. Wonder if they will add some of that to the mobile app. It already has trim.

**Comment by upsert** [comment]
Firmware 06.11.04f000 dropped. No change.

**Comment by upsert** [comment]
Firmware 06.12.01f000 is out. I expect it will be for the [motion sensor and repeater](http://www.lutron.com/en-US/Company-Info/Pages/News/Media-PressCenter/PressReleases/PressReleaseDetail.aspx?prid=796) they just added to Caseta. Unfortunately that does not mean motion sensors over telnet have changed.

**Comment by czoog** [comment]
> Firmware 06.12.01f000 is out. I expect it will be for the [motion sensor and repeater](http://www.lutron.com/en-US/Company-Info/Pages/News/Media-PressCenter/PressReleases/PressReleaseDetail.aspx?prid=796) they just added to Caseta. Unfortunately that does not mean motion sensors over telnet have changed.

Well fingers crossed that they add it!  In any event it's good to see them adding motion sensors!

**Comment by upsert** [comment]
Firmware 08.00.06f000 is out. Big version number jump, so maybe this one is the occ/vac sensor and repeater for Caseta. Apparently the motion sensor is not supported on HomeKit either, but maybe if they add it we could get it on telnet too.

**Comment by zwily** [comment]
@upsert Where do you see these new firmware releases? And are they applied automatically to our hubs? (Sorry to go a little off-topic, I’ve been using RA2 Select for a couple years and have always wondered this and never found anything.)

**Comment by upsert** [comment]
@zwily The hub broadcasts a status including the firmware version using mDNS and DNS-SD and I periodically use the [Bonjour Browser](https://hobbyistsoftware.com/bonjourbrowser) to check it.

**Comment by bdubsw** [comment]
@upsert have you got your hands on a motion sensor yet and tested it out?

**Comment by upsert** [comment]
@bdubsw Yes, since 2018 on the Ra2 Select line. They are using the same hardware for Caseta.

**Comment by BWilky** [comment]
Was just curious if the PD-OSENS-WH is exposed over telnet like everything else and sends motion events

**Comment by dwest576** [comment]
PD-OSENS-WH is NOT exposed over telnet

**Comment by morph166955** [comment]
I've been researching this same issue for another application and stumbled on this bug.  Perhaps my info may help you.

I believe the answer here is in the JSON side of the protocol.  Please look at https://github.com/hassio-addons/addon-lutron-cert to see how to create the necessary 3 certs/keys.  You can validate that you can receive a stream with those files using "curl -k --cacert caseta-bridge.crt --key caseta.key --cert caseta.crt https://bridge-ip:8081"

From there I would suggest looking at https://github.com/gurumitts/pylutron-caseta/commit/c14487c7d8a71af36b5958d324e46b5ca0509215 for examples of JSON you can send to query about the area.  

My theory, which I haven't been able to validate/test, is that you could regularly query through the JSON the occupancy state of a room/area.  Everything I can tell from my testing makes it seem like the occupancy sensor works like a pico remote as opposed to an actual switch.

I hope this helps.

**Comment by upsert** [comment]
It may be possible over HTTPS/JSON, but I will leave that to the Caseta integration to reverse engineer the protocol.

This integration is only for the Telnet-based Lutron Integration Protocol. The hope is that since Telnet is the method for professionals to integrate with Caseta Pro and RA2 Select (and virtually all the Lutron main product lines), that they will eventually add it to the Telnet protocol and bring it up to par with RadioRa2.

On a related note, firmware 08.01.12f000 is out and sadly, it does not change the Telnet output.

**Comment by contactcr** [comment]
I guess it got added to the other protocol, wonder why Telnet is (still) absent.

https://www.home-assistant.io/integrations/lutron_caseta/#sensor

**Comment by upsert** [comment]
Firmware 08.01.13f000 is out. No changes to the telnet interface.

**Comment by boomalator** [comment]
I have a Caseta Occupancy Sensor, too, and was hoping it would be available :(

I notice that the default Caseta integration says that it supports occupancy sensors... but it doesn't give Pico events.

Since I have a non-pro hub as well, I suppose I can run both integrations with only the occupancy sensor as the only device connected to the one....

**Comment by boomalator** [comment]
Oh, and for what it's worth... it does show up in the integration report.

**Comment by psperry** [comment]
I met a guy online who works at Lutron and he offered to send my request to add support to the correct team if I email it to him. Do you think that’s a good idea? If so, any advise on how the request should be worded?

**Comment by cjmaio** [comment]
Exposing this to the telnet interface is the important bit, and would make those sensors infinitely more useful especially in situations where you have more than just Lutron devices.

**Comment by boomalator** [comment]
Well, I think it’s a great idea. You can let your friend know that although I have many Lutron products (including as far back as RadioRa “Classic”) I can’t/won’t buy their occupancy sensors if they can’t be integrated. I have two useless sensors (the Caseta, and a RA2 one that I received as a promo…. Which is just enough different to require a whole new hub. ☹ )

**Comment by psperry** [comment]
Well, I will give it a shot. No promises of course, but it can’t hurt to ask.

**Comment by upsert** [comment]
Looks like there will be a new Ra2 Select and HomeWorks hub coming soon: https://fcc.report/FCC-ID/JPZ0129

Same hardware for both and each has a repeater variation. Bluetooth Low Energy (BLE), 802.15.4 wireless (likely for Ketra) and 431 MHz ClearConnect. Uses 48 Volt Power-over-Ethernet which would be nice to have on the current hubs. Model numbers **HQP7-RF-2**, **HQREP-RF**, **RR-PROC3**, and **RR-REP3**.

**Comment by scottd83** [comment]
Good find! Finally some love for RA2 Select! The label pics look like the existing "Clear Connect Gateway - Type X" that they have for Homeworks and Athena. 

I bailed on Caseta for RA2 Select because it was missing Occupancy / Vacancy Sensors, had a low device limit (50 at the time), and didn't have the 15A appliance model. And soon it looks like Caseta will have all the things I left for, was starting to get a little disappointed that RA2 Select wasn't getting any new or cool features. Still holding out for them to unlock fade times, etc. (outside of scenes) and sensor change of state over telnet. 

edit: Oh and thanks for constantly monitoring and updating us on changes!

**Comment by gmcmicken** [comment]
The latest Lutron integration protocol guide has a lot of references to occupancy monitoring. Has it shown up on the Caseta Smart Bridge Pro yet? Rev. Dec 16

https://www.lutron.com/TechnicalDocumentLibrary/040249.pdf

**Comment by dcode** [comment]
Another note to follow up on @upsert's analysis, looks like the FCC specifically did Zigbee and BLE testing. Will have to wait until the full release, but maybe they'll be able to control Hue (like their Aurora dimmer)! I mean, I'm not holding my breath, but that'd be amazing. Give me Hue or Ketra or both!

**Comment by upsert** [comment]
Another curious device from Lutron:
https://fcc.report/FCC-ID/JPZ0136/

Bluetooth Low Energy **Darter** keypad. Described as a "... wireless remote keypad for installation into a standard WD wallbox and powered at 120-277V. Device does not have a load but provides wireless signals to a Lutron lighting control system." Operates at 2.4 GHz, but nothing at 431 MHz for ClearConnect. Model numbers look like Pico model numbers with 3 and 4-button variants but `RRST` and `HRST` prefix.

**Comment by scottd83** [comment]
Looks like those devices are all for the newly launched RA3:

https://radiora3.lutron.com/

**Comment by zwily** [comment]
Was this closed because it works now? Or just won’t ever be implemented because the hardware doesn’t support it?
